### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by MVrachev**
As we discussed before #227
there are many benefits to creating a Docker image which will be run
on Heroku. That way we would be free from the limitations provided
by Heroku and we would be able to use Go modules.
Also, if one day we decide to move Precaution on another
hosting platform it will be really easy.

We won't need the ".profile" file anymore because we would install
Gosec in our Dockerfile as all other dependencies.

Closes: https://github.com/vmware/precaution/issues/227

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+RUN curl -OL https://github.com/securego/gosec/releases/download/2.0.0/gosec_2.0.0_linux_amd64.tar.gz
+
+# We need sha256sum which is not installed by default in toybox but it's available in util-linux.
+# There is a conflict between toybox and util-linux and that's why we have to remove toybox.
+RUN tdnf remove toybox -y && tdnf install util-linux -y
+
+# Validate Gosec tar.gz file checksum
+RUN echo "490c2a0434b2b9cbb2f4c5031eafe228023f1ac41b36dddd757bff9e1de76a2b  ./gosec_2.0.0_linux_amd64.tar.gz" | sha256sum -c -
+
+# Install all nessery tools to unzip the gosec tar.gz file
```
Comment (joshuagl) [inline Dockerfile]: > nessery -> necessary

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Cleanup of the already unnecessary installed packages
+RUN tdnf remove -y \
+    python3-pip \
+    python3-setuptools \
+    git \
+    util-linux \
+    tar \
+    gzip
+
+# Start application on port given by Heroku
```
Comment (ericwb) [inline Dockerfile]:
```suggestion
# Start the application on the port given by Heroku
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+# Install all necessary tools to unzip the gosec tar.gz file
+RUN tdnf install -y \
+    tar \
+    gzip
+
+# Unzip and remove the redundant tar.gz file
+RUN tar -xvzf gosec_2.0.0_linux_amd64.tar.gz -C /usr/bin && rm ./gosec_2.0.0_linux_amd64.tar.gz
+
+# Cleanup of the already unnecessary installed packages
```
Comment (ericwb) [inline Dockerfile]:
```suggestion
# Cleanup of unnecessary installed packages
```

File: merge_reports.js [hunk 4]
```diff
@@ -31,7 +31,10 @@ const mergeSummaries = (summaries) => {
  * for object schema see: https://developer.github.com/v3/checks/runs/#output-object
  */
 module.exports = (reports) => {
-  const annotations = reports.map(r => r.annotations).flat()
+  let annotations = []
```
Comment (ericwb) [inline merge_reports.js]: This change is unrelated and should not be part of this PR.

Comment (joshuagl) [reply merge_reports.js]: This change was merged as part of PR #247
